src/k8s/pkg/component/hook.go

```diff
@@ -26,11 +27,7 @@ func networkValues() (map[string]any, error) {
 	}
 
 	// TODO: the cluster cidr should be configurable through a common interface
-	clusterCIDRStr, err := utils.GetServiceArgument("kube-proxy", "--cluster-cidr")
-	if err != nil {
-		return nil, fmt.Errorf("failed to get cluster cidrs from kube-proxy arguments: %w", err)
-	}
-
+	clusterCIDRStr := snap.GetServiceArgument(snap.Instance(), "kube-proxy", "--cluster-cidr")
```

Comment:
Ideally, `snap.Instance()` would eventually be initialized once and passed to all callers as needed, but it's OK to start with this.

Reply:
I agree. However, this is not yet easily possible with the microcluster framework as described in the GH comment. We'll can update this once microcluster supports something like `OnRequest` hooks (which is "in the making" according to the microcluster folks).